File: Android_Zone_Test/src/com/example/mylib_test/activity/three_place/FrescoActivity.java
package com.example.mylib_test.activity.three_place;

import android.content.Context;
import android.net.Uri;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import com.example.mylib_test.R;
import com.example.mylib_test.delegates.FrescoDeletates;
import com.zone.adapter3.QuickRcvAdapter;
import com.zone.adapter3.base.IAdapter;
import com.zone.lib.base.activity.BaseActivity;
import java.util.ArrayList;
import java.util.List;
import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Created by Administrator on 2016/4/4.
 */
public class FrescoActivity extends BaseActivity {
    @Bind(R.id.rv)
    RecyclerView rv;

    private List<Entity> mDatas = new ArrayList<>();
    private IAdapter<Entity> muliAdapter;

    public class Entity {
        public String introduce;
        public String uri;

        public Entity(String introduce, String uri) {
            this.introduce = introduce;
            this.uri = uri;
        }
    }


    @Override
    public void setContentView() {
        setContentView(R.layout.a_fresco);
        ButterKnife.bind(this);
    }

    @Override
    public void findIDs() {

    }

    @Override
    public void initData() {
        //todo   HTTP,File,Resource,Uri,Gif,Mp4,Error;
        mDatas.add(new Entity("http","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        mDatas.add(new Entity("File","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        mDatas.add(new Entity("Resource","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        mDatas.add(new Entity("Uri","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        mDatas.add(new Entity("Gif","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        mDatas.add(new Entity("Mp4","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        mDatas.add(new Entity("Error","https://raw.githubusercontent.com/facebook/fresco/gh-pages/static/logo.png"));
        rv.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        rv.setItemAnimator(new DefaultItemAnimator());
        muliAdapter = new QuickRcvAdapter(this, mDatas) {
//            @Override
//            protected int getItemViewType2(int dataPosition) {
//                return dataPosition % 3 == 0 ? 1 : 0;
//            }
        };
        muliAdapter
                .addViewHolder(new FrescoDeletates())//默认
                .relatedList(rv)
                .addItemDecoration(10);
    }

    @Override
    public void setListener() {

    }

    @Override
    public void onClick(View v) {
        super.onClick(v);
    }

    public static final String ANDROID_RESOURCE = "android.resource://";
    public static final String FOREWARD_SLASH = "/";

    private static Uri resourceIdToUri(Context context, int resourceId) {
        return Uri.parse(ANDROID_RESOURCE + context.getPackageName() + FOREWARD_SLASH + resourceId);
    }
}

File: Android_Zone_Test/src/com/example/mylib_test/activity/three_place/FrescoActivity.kt
package com.example.mylib_test.activity.three_place

import android.graphics.Rect
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.mylib_test.R
import com.example.mylib_test.activity.db.entity.MenuEntity
import com.zone.lib.base.controller.activity.BaseFeatureActivity
import com.example.mylib_test.adapter.delegates.FrescoDeletates
import com.example.mylib_test.adapter.delegates.FrescoProcessorDealDeletates
import com.example.mylib_test.adapter.delegates.FrescoProcessorDeletates
import com.example.mylib_test.adapter.delegates.FrescoSuperscriptDeletates
import com.zone.adapter3kt.QuickAdapter
import com.zone.adapter3kt.ViewStyleDefault
import com.zone.adapter3kt.ViewStyleOBJ
import kotlinx.android.synthetic.main.a_fresco.*

/**
 *[2018/7/8] by Zone
 */
class FrescoActivity : BaseFeatureActivity() {

    inner class Entity(var introduce: String, var uri: String);

    private var mDatas = ArrayList<Entity>()

    init {
        mDatas.add(Entity("http", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("File", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("Resource", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("Uri", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("Gif", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("Mp4", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("Error", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("角标", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("进度", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
        mDatas.add(Entity("处理", "http://ww4.sinaimg.cn/mw1024/005PquKVgw1ezrgh5ppeyj30ku0kujvv.jpg"))
    }

    override fun setContentView() {
        setContentView(R.layout.a_fresco)
    }


    override fun initData() {
        rv!!.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        rv!!.itemAnimator = DefaultItemAnimator()
        rv.adapter =   QuickAdapter<Entity>(this@FrescoActivity).apply {
            registerDelegate(FrescoDeletates())
            registerDelegate(1, FrescoSuperscriptDeletates())
            registerDelegate(2, FrescoProcessorDeletates())
            registerDelegate(3, FrescoProcessorDealDeletates())
            setStyleExtra(object : ViewStyleDefault<Entity>() {
                override fun generateViewStyleOBJ(item: Entity): ViewStyleOBJ? {

                    var bottom = 10
                    val viewStyle = when (item.introduce) {
                        "角标" -> 1
                        "进度" -> 2
                        "处理" -> {
                            bottom = 0
                            3
                        }
                        else -> -1
                    }

                    return ViewStyleOBJ().viewStyle(viewStyle)
                            .divderRect(Rect(10, 0, 10, bottom))

                }

                override fun getItemViewType(position: Int, itemConfig: ViewStyleOBJ) {
                }
            })
            add(mDatas)
        }

    }

    override fun setListener() {
    }
}